lib/krill/show_response.rb

```diff
@@ -0,0 +1,79 @@
+module Krill
+
+  # @api krill
+  require 'delegate'
+  class ShowResponse < SimpleDelegator
+
+    # Return the response that was stored under `var`. When used with 
+    # a key associated with a table response-set, returns a list of table responses
+    # in order of the rows of the table.
+    #
+    # @param var [Symbol/String]  The var used to store data under
+    #               specified in the get or select call of the ShowBlock
+    def get_response var
+      responses[var.to_sym]
+    end
+
+    # Returns data recorded in a specified row of an input table
+    # @param var [Symbol/String]  the table key specified to store data under
+    #               in the get.
+    # @param opts [Hash]  additional options
+    # @option op [Operation]  the Operation for which to retrieve the table data
+    #               for, supposing the table was made on an OperationsList
+    # @option row [Integer]  the row index of the table for which to retrieve table data
+    # @return [String/Fixnum]  the data inputted in the particular input cell specified
+    #               by the column associated with `var`, 
+    #               and the row associated with either `op` or `row`   
+    #               returns nil if the requested row/column pair doesn't exist
+    def get_table_response var, opts = {}
+      if (opts[:op] && opts[:row]) || (!opts[:op] && !opts[:row])
+        raise TableCellUndefined, "Invalid parameters for get_table_response - specify one of op or row, not both"
+      elsif self[:table_inputs].nil?
+        return nil
+      elsif opts[:op]
+        if self[:table_inputs].first[:opid] < 0
+          raise TableCellUndefined, "Invalid parameters for get_table_response - an :op option cannot be specified for a table that doesn't have operations corresponding to its rows"
+        end
+        opid = Operation.find(opts[:op]).id # return op.id if passed an operation or the id itself
+        target_table_input = self[:table_inputs].find { |ti| (ti[:key].to_sym == var.to_sym) && (ti[:opid] == opid) }
```

Comment:
Factor out lines 38 and 40 by determining whether you are using `:op` or `:row` and setting an index variable